extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -151,6 +160,41 @@ class MongoQueueProcessor {
         return this._consumer.close(done);
     }
 
+    _isMasterKey(entry) {
+        return entry.getObjectVersionedKey() === entry.getObjectKey();
+    }
+
+    _getDataContent(entry) {
+        const contentLength = entry.getContentLength();
+        if (contentLength > 0) {
+            return ['DATA', 'METADATA'];
+        } else {
+            return ['METADATA'];
+        }
+    }
+
+    _getZenkoObjectMetadata(entry, done) {
+        const bucket = entry.getBucket();
+        const key = entry.getObjectKey();
+        const params = { versionId: entry.getVersionId() };
+
+        this._mongoClient.getObject(bucket, key, params, this.logger,
+        (err, data) => {
+            if (err && !err.NoSuchKey) {
+                this.logger.error('error getting zenko object metadata', {
+                    method: 'MongoQueueProcessor._getZenkoObjectMetadata',
+                    error: err,
+                    entry: entry.getLogInfo(),
+                });
+                return done(err);
+            }
+            if (err && err.NoSuchKey) {
```

Comment:
Minor suggestion: maybe reads a bit cleaner if this is the first error condition—that way you can simplify the above condition as `if (err)`.